docs/interfaces.md

```diff
@@ -0,0 +1,568 @@
+# FPC Component Interfaces and States
+
+This document defines the interfaces exposed by the FPC components and their internal state.
+
+
+# ERCC
+
+## Interface:
+The ERCC interface is implemented through a "normal" chaincode interface. All functions can be called by admin or via cc2cc.
+
+```go
+// chaincode interface (exposed to ecc and clients)
+func Invoke(stub shim.ChaincodeStubInterface) pb.Response {}
+
+// functions below are dispatched by Invoke implementation
+// Note: Maybe use protobuf-to-json mapping (https://developers.google.com/protocol-buffers/docs/proto3#json)
+// to map the methods and their arguments into to chaincode
+// invocation form { "Function": "..", "Args": [....]}
+
+// returns a set of credentials registered for a given chaincode id
+func queryCCInfos(chaincode_id string) (chaincode_ek []byte, allCredentials []Credentials) {}
```

Comment:
not so sure about this. You are proposing a `queryRegisteredEnclaves(chaincode_id string)` that then just returns the public keys. that's fine and would work for MVP. 

I thought `queryCCInfos` is more a general purpose query to get all information available for a certain FPC chaincode. As the credentials now also contains some enclave host identifier, it is also useful to pick a set of endorsing enclaves. At least this is what I had in mind.

To address 2., I suggest that we hide these protos as much as possible. Our FPC client SDK layer still should have no problems dealing with this. In case a client wants this data, we could present it as json format as @g2flyer suggested in this conversation.

Reply:
> You are proposing a queryRegisteredEnclaves(chaincode_id string)

That's right, in the end... we call it "enclave registry"!!

Anyway, no strong opinion about the arguments above.
So, resolve this and continue a related conversation in [this comment](https://github.com/hyperledger-labs/fabric-private-chaincode/pull/420#discussion_r468235021).